frontend/src/components/pages/Scheduling/SelectDateTime.tsx

```diff
@@ -245,9 +272,73 @@ const SelectDateTime = ({
         value: recurringDate.toString(),
       },
     });
+    setFormErrors({
+      ...formErrors,
+      recurringDonationEndDate: "",
+    });
+  };
+
+  const validateForm = () => {
+    const newErrors = {
+      date: "",
+      dayPart: "",
+      timeRange: "",
+      frequency: "",
+      recurringDonationEndDate: "",
+    };
+    let valid = true;
+
+    if (!date || date.toString() === "Invalid Date") {
+      valid = false;
+      newErrors.date = ErrorMessages.requiredField;
+    }
+    if (!dayPart) {
+      valid = false;
+      newErrors.dayPart = ErrorMessages.requiredField;
+    }
+    if (!endTime) {
+      // Null endTime means a timeRange was not selected in the form
+      valid = false;
+      newErrors.timeRange = ErrorMessages.requiredField;
+    }
+    if (!frequency) {
+      valid = false;
+      newErrors.frequency = ErrorMessages.requiredField;
+    } else if (frequency !== "One time donation") {
+      if (!recurringDonationEndDate) {
+        valid = false;
+        newErrors.recurringDonationEndDate = ErrorMessages.requiredField;
+      } else if (recurringDonationEndDate === "Invalid Date") {
+        valid = false;
+        newErrors.recurringDonationEndDate =
+          ErrorMessages.invalidRecurringDonationEndDateFormat;
+      } else {
+        // Validate end date is within 6 months of start date
+        const startDate = new Date(startTime);
+        const maxEndDate = new Date(startTime);
+        maxEndDate.setMonth(startDate.getMonth() + 6);
+        const endDate = new Date(recurringDonationEndDate);
+        if (!(startDate <= endDate && endDate <= maxEndDate)) {
```

Comment:
I think we should display a different error message when startDate > endDate, smth like `Recurring end date must be after the selected start date`

Reply:
Also if its being edited, can we not check whether end date is less than maxEndDate?